async/runtime.mli

```diff
@@ -35,6 +35,7 @@ val run_request :
      region:string
   -> access_key:string
   -> secret_key:string
+  -> ?token:string
```

Comment:
It seems that a request can have either `token` or a `access_key/secret_key`, the type on this function should indicate this. Something like:

``` ocaml
type security_token = 
 | AccessKey string string
 | Token string
```
would be an improvement.

Could you also update the comment associated with this function to mention it now supports a token?
I've got document generation available at http://inhabitedtype.github.io/ocaml-aws/

Reply:
So the token works by providing an additional token on top of a key/secret. The flow being that you call sts assume-role or some other endpoint, get a new key, secret, and token. You can then use all three of those to authenticate requests until the credentials expire.

There is a bit more information here https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_use-resources.html which covers some of the main points but a bunch of the information is spread across a wide range of their docs.

Reply:
Got it, thanks for the explanation. 
Happy with the code as it stands in that case.